From e5538784bf0253632c5f4e1873bac03727e94daf Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 07 Oct 2019 11:30:05 +0300
Subject: [PATCH] [SW-0] goya: remove redundant code and update relevant comment

This code is not needed
1) prstn is tied to rstn so no need to wait for prstn since it is up
once we start running
2) FW FSM is already set at BTL no need to do it again

Change-Id: Ifd4767e2f367d74066743b8e111211142426c539
---

diff --git a/goya/common.c b/goya/common.c
index ef65759..2b65955 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -75,18 +75,14 @@
 	}
 }
 
-/* Handle PCIe erratum: make sure reset_n de-asserted only after perst */
+/*
+ * Handle PCIe erratum: make sure reset_n de-asserted only after perst
+ * Assume PRSTn tied with cold reset
+ * Assume PCIe enable strap (DIP switch) is set, and hence HW enable LTSSM
+ */
 void preboot_pcie_wait_for_link(void)
 {
 	preboot_wait_for_ext_ld_done();
-
-	/* Wait for PRSTn */
-	while (!regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN))
-		;
-
-	/* Enable FW FSM after PRSTn */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PCI_FW_FSM);
-
 	preboot_wait_for_stable_pclk();
 
 	while (1) {
